### DISCUSSION TITLE
differences bt sapien and robosuite on mujoco

### DISCUSSION THREAD

**Discussion by shivanimall** [haosulab/SAPIEN#75] [Q&A] [answered]
Thank you for making this available, it is great work! 

I am wondering what are some major differences you've noticed between Sapien vs using robosuite framework on mujoco?

Thanks for your help.

**Comment by fbxiang** [comment, marked as answer]
The major difference are the following
1. SAPIEN uses PhysX4 as backend while robosuite uses mujoco. So SAPIEN allows adding, modifying, and removing objects at run time while robosuite does not. On the other hand, mujoco may support more solvers and has more tunable parameters.
2. SAPIEN provides a feature-rich and customizable rasterizer and a ray tracer. So on the rendering side, SAPIEN can do a lot more.
3. Robosuite offers other functionalities such as controllers, robots, tasks, and environments, which we do not provide in SAPIEN, but we provide them in downstream software and applications that use SAPIEN, such as ManiSkill.

**Reply by shivanimall** [reply]
Many thanks for sharing!
